### PR TITLE
feat: snap types restructure

### PR DESCRIPTION
**Pull request by tadejpodrekar**
Moved reusable types from the connector and snap to the types package.
Corrected imports to use types from packages.

### CODE REVIEW

File: packages/types/methods.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+export interface SetVCStore {
+  method: 'setVCStore';
+  params: {
+    vcStore: string;
+  };
+}
+
+export interface GetAvailableVCStores {
+  method: 'getAvailableVCStores';
+}
```
Comment (andyv09) [inline packages/types/methods.ts:73]: Update these methods to match the ones from snap (use interfaces for params from ./params)

File: packages/types/requests.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  | GetVP
+  | ChangeInfuraToken
+  | TogglePopups
+  | GetDID
+  | GetMethod
+  | GetAvailableMethods
+  | SwitchMethod
+  | GetVCStore
+  | SetVCStore
+  | GetAvailableVCStores;
```
Comment (andyv09) [inline packages/types/requests.ts:28]: change with latest methods from snap

File: packages/types/snapApi.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  changeInfuraToken(infuraToken: string): Promise<boolean>;
+  togglePopups(): Promise<boolean>;
+  getDID(): Promise<string>;
+  getMethod(): Promise<string>;
+  getAvailableMethods(): Promise<string[]>;
+  switchMethod(didMethod: string): Promise<boolean>;
+  getVCStore(): Promise<string>;
+  setVCStore(vcStore: string): Promise<boolean>;
+  getAvailableVCStores(): Promise<string[]>;
+}
```
Comment (andyv09) [inline packages/types/snapApi.ts:24]: update this aswell

**Review by andyv09 — CHANGES_REQUESTED** [review]

File: packages/types/methods.ts [hunk 4]
```diff
@@ -1,37 +1,43 @@
...
 export interface VCQuery {
   [key: string]: string;
 }
 
-export interface GetVCs {
-  method: 'getVCs';
-  params: {
-    query?: VCQuery;
-  };
+export interface QuerryVCs {
```
Comment (andyv09) [inline packages/types/methods.ts:8]: Query*

**Review by andyv09 — CHANGES_REQUESTED** [review]

**Review by andyv09 — APPROVED** [review]

File: packages/types/constants.ts [hunk 5]
```diff
@@ -18,6 +18,20 @@ export const didCoinTypeMappping: Record<string, number> = {
   'did:key': 60,
 };
 
+// Source: https://github.com/multiformats/multicodec/blob/master/table.csv
```
Comment (martines3000) [inline packages/types/constants.ts:21]: We don't need to split this. You can leave it in snap.

File: packages/types/params.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+export type SaveVCRequestParams = {
+  verifiableCredential: W3CVerifiableCredential;
+  options?: SaveVCOptions;
+};
+
+export type DeleteVCRequestParams = {
+  id: string | [string];
+  options?: {
```
Comment (martines3000) [inline packages/types/params.ts:48]:
```ts
type DeleteVCOptions = {
    store?: AvailableVCStores | [AvailableVCStores];
}
```

File: packages/types/params.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  vcs: [
+    {
+      id: string;
+      metadata?: {
+        store?: AvailableVCStores;
+      };
+    }
+  ];
+  proofFormat?: SupportedProofFormats;
+  proofOptions?: {
```
Comment (martines3000) [inline packages/types/params.ts:18]: Maybe split to separate type `proofOptions`

File: packages/types/constants.ts (hunk 5 above)
Comment (martines3000) [reply packages/types/constants.ts:21]: fixed

**Pull request merged**